### PR TITLE
[E2E] Overload the existing E2E `visitDashboard` helper to make it work with the alias

### PR DESCRIPTION
**Pull request by nemanjaglumac**
This PR overloads the existing `visitDashboard` E2E helper to make it accept either:
1. a direct dashboard id (number) or
2. a dashboard id alias (string)

It then replaces and updates related specs that were grabbing the dashboard id alias and then calling the `visitDashboard` function. The code can be simplified like this:

```diff
- cy.get("@emptyDashboard").then(id => {
-   visitDashboard(id);
- });
+ visitDashboard("@emptyDashboard");
```

### CODE REVIEW

File: e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js [hunk 1]
```diff
@@ -675,7 +675,12 @@ const openSlowDashboard = (params = {}) => {
     "cardQuery",
   );
 
-  visitDashboard("@dashboardId", { params });
+  cy.get("@dashboardId").then(dashboardId => {
+    return cy.visit({
+      url: `/dashboard/${dashboardId}`,
+      qs: params,
+    });
+  });
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js:683]: For some reason (that I couldn't figure out), every test that uses `cy.clock` in combination with the `visitDashboard` helper fails. It works only when `cy.visit()` is explicitly used.

If anyone has ideas why, I'm all ears.

Comment (nemanjaglumac) [reply e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js:683]: An example of a failed run:
https://github.com/metabase/metabase/actions/runs/7761351435/job/21214199066#step:11:783

And the recording of that failure:
https://app.replay.io/recording/b9a3ad12-f5ff-465b-98be-8a9813df8aa6

**Review by kamilmielnik — APPROVED** [review]
:+1:

File: e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js [hunk 2]
```diff
@@ -199,9 +199,7 @@ describe("embed modal display", () => {
   describeEE("when the user has a paid instance", () => {
     it("should display a link to the interactive embedding settings", () => {
       setTokenFeatures("all");
-      cy.get("@dashboardId").then(id => {
-        visitResource("dashboard", id);
-      });
+      visitDashboard("@dashboardId");
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js:202]: I think it still does in a lot of places
https://github.com/metabase/metabase/blob/d24e534322c9c1345afd1804ea3197085f90a56b/e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js#L36

**Pull request merged**